File: platform/src/main/java/com/toocol/termio/platform/component/IStyleAble.java
package com.toocol.termio.platform.component;

import javafx.scene.Node;

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/8/11 14:00
 */
public interface IStyleAble {
    String[] styleClasses();

    /**
     * add styles to components
     */
    default void styled() {
        if (!(this instanceof Node)) {
            return;
        }
        ((Node) this).getStyleClass().addAll(styleClasses());
    }
}

File: platform/src/main/java/com/toocol/termio/platform/component/IStyleAble.kt
package com.toocol.termio.platform.component

import javafx.scene.Node

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/8/11 14:00
 */
interface IStyleAble {
    fun styleClasses(): Array<String>

    /**
     * add styles to components
     */
    fun styled() {
        if (this !is Node) {
            return
        }
        (this as Node).styleClass.addAll(*styleClasses())
    }
}